doc/src/python.rst

```diff
@@ -70,80 +77,101 @@ Examples
+operate more like a true programming language.
 
 Use of this command requires building LAMMPS with the PYTHON package
 which links to the Python library so that the Python interpreter is
 embedded in LAMMPS.  More details about this process are given below.
 
 There are two ways to invoke a Python function once it has been
-defined.  One is using the *invoke* keyword.  The other is to assign
+registered.  One is using the *invoke* keyword.  The other is to assign
 the function to a :doc:`python-style variable <variable>` defined in
-your input script.  Whenever the variable is evaluated, it will
-execute the Python function to assign a value to the variable.  Note
-that variables can be evaluated in many different ways within LAMMPS.
-They can be substituted for directly in an input script.  Or they can
-be passed to various commands as arguments, so that the variable is
-evaluated during a simulation run.
-
-A broader overview of how Python can be used with LAMMPS is given on
-the :doc:`Python <Python_head>` doc page.  There is an examples/python
-directory which illustrates use of the python command.
+your input script.  Whenever the variable is evaluated, it will execute
+the Python function to assign a value to the variable.  Note that
+variables can be evaluated in many different ways within LAMMPS.  They
+can be substituted with their result directly in an input script, or
+they can be passed to various commands as arguments, so that the
+variable is evaluated during a simulation run.
+
+A broader overview of how Python can be used with LAMMPS is given in the
+:doc:`Use Python with LAMMPS <Python_head>` section of the
+documentation.  There also is an ``examples/python`` directory which
+illustrates use of the python command.
 
 ----------
 
-The *func* setting specifies the name of the Python function.  The
-code for the function is defined using the *file* or *here* keywords
-as explained below. In case of the *source* keyword, the name of
-the function is ignored.
+The first argument of the *python* command is either the *source*
+keyword or the name of a Python function.  This defines the mode
+of the python command.
+
+If the *source* keyword is used, it is followed by either a file name or
+the *here* keyword.  No other keywords can be used.  The *here* keyword
+is followed by a string with python commands, either on a single line
+enclosed in quotes, or as multiple lines enclosed in triple
+quotes.  These Python commands will be passed to the python interpreter
+and executed immediately without registering a Python function for
+future execution.  This allows to run arbitrary Python code at any
+time while processing the LAMMPS input file.  This can be used to pre-load
+Python modules, initialize global variables or perform operations
+using the python programming language.  The Python code will be executed
+in parallel on all MPI processes.  No arguments can be passed.
```

Comment:
my higher level comment on this paragraph is to explain that the results of "source" mode
persist in the Python interpreter so that later use of the python command (or a python variable) can use them - not 100% clear that is correct, but I think that what you told me.
If so, I think it's a key motivation for using "source" mode, so should be explained here.

Reply:
> my higher level comment on this previous paragraph is to explain that the results of "source" mode persist in the Python interpreter so that later use of the python command (or a python variable) can use them - not 100% clear that is correct, but I think that what you told me.
> 
> If so, I think it's a key motivation for using "source" mode, so should be explained here.

This is true for *any* of the methods that execute python code. The motivation for the "source" keyword is that you want to execute python code *without* registering a python function for future use (with a python style variable or the python invoke command or fix python/*). I've made some minor tweaks with the intent to improve clarity, but I don't see anything messing. It already feels a bit redundant.